content/articles/how-to-implement-k-fold-cross-validation/index.md

```diff
@@ -0,0 +1,417 @@
+'Mean Training Recall': 0.9881935259310826, 
+'Training F1 scores': array([0.99104478, 0.99104478, 1.        , 0.99408284, 0.99115044]), 
+'Mean Training F1 Score': 0.9934645669906736, 
+'Validation Accuracy scores': array([0.92105263, 0.93859649, 0.94736842, 0.93859649, 0.94690265]), 
+'Mean Validation Accuracy': 93.85033379909953, 
+'Validation Precision scores': array([0.925     , 1.        , 0.90909091, 0.88888889, 0.89130435]), 
+'Mean Validation Precision': 0.9228568291611768, 
+'Validation Recall scores': array([0.86046512, 0.8372093 , 0.95238095, 0.95238095, 0.97619048]), 
+'Mean Validation Recall': 0.9157253599114064, 
+'Validation F1 scores': array([0.89156627, 0.91139241, 0.93023256, 0.91954023, 0.93181818]), 
+'Mean Validation F1 Score': 0.9169099279932611}
+```
+
+Let us visualize the results of the second model.
+
+```python
+# Plot Accuracy Result
+plot_result(model_name,
+            "Accuracy",
+            "Accuracy scores in 5 Folds",
+            decision_tree_result_2["Training Accuracy scores"],
+            decision_tree_result_2["Validation Accuracy scores"])
+```
+
+![Accuracy Plot of the Second Decision Tree Model](/engineering-education/how-to-implement-k-fold-cross-validation/accuracy-2.png)
+
+```python
+# Plot Precision Result
+plot_result(model_name,
+            "Precision",
+            "Precision scores in 5 Folds",
+            decision_tree_result_2["Training Precision scores"],
+            decision_tree_result_2["Validation Precision scores"])
+```
+
+![Precision Plot of the Second Decision Tree Model](/engineering-education/how-to-implement-k-fold-cross-validation/precision-2.png)
+
+```python
+# Plot Recall Result
+plot_result(model_name,
+            "Recall",
+            "Recall scores in 5 Folds",
+            decision_tree_result_2["Training Recall scores"],
+            decision_tree_result_2["Validation Recall scores"])
+```
+
+![Recall Plot of the Second Decision Tree Model](/engineering-education/how-to-implement-k-fold-cross-validation/recall-2.png)
+
+```python
+# Plot F1-Score Result
+plot_result(model_name,
+            "F1",
+            "F1 Scores in 5 Folds",
+            decision_tree_result_2["Training F1 scores"],
+            decision_tree_result_2["Validation F1 scores"])
+```
+
+![F1-Score Plot of the Second Decision Tree Model](/engineering-education/how-to-implement-k-fold-cross-validation/f1-score-2.png)
+
+We can see that the validation results of the second model in each fold are better. It has a mean validation accuracy of 93.85% and a mean validation f1 score of 91.69%.
+
+
+### Conclusion
+When training a model on a small data set, the K fold cross-validation technique comes in handy. You may not need to use K fold cross-validation if your data collection is huge. The reason is you have enough records in your validation set to check the machine learning model.  It takes a lot of time to use K fold cross-validation on a large data collection. 
```

Comment:
Make sure `k-fold` has a hyphen in between. Go through the whole tutorial and edit accordingly

Reply:
done